lib/ansible/template/__init__.py

```diff
@@ -348,7 +348,7 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
             else:
                 # Do we need to convert these into text_type as well?
                 # return self._clean_data(to_text(variable._obj, nonstring='passthru'))
-                rval = self._clean_data(variable._obj)
+                rval = self._clean_data(variable)
```

Comment:
This looks like we should now call _clean_data(variable) unconditionally, no need for the if isinstance check anymore.